akka-cluster-tools/src/main/scala/akka/cluster/pubsub/DistributedPubSubMediator.scala

```diff
@@ -666,14 +666,15 @@ class DistributedPubSubMediator(settings: DistributedPubSubSettings) extends Act
       if (matchingRole(m))
         nodes += m.address
 
-    case MemberRemoved(m, _) ⇒
-      if (m.address == selfAddress)
-        context stop self
-      else if (matchingRole(m)) {
+    case MemberLeft(m, _) ⇒
+      if (matchingRole(m)) {
         nodes -= m.address
         registry -= m.address
       }
 
+    case MemberRemoved(m, _) if m.address == selfAddress ⇒
```

Comment:
the `if` should be after `=>` otherwise it will be "logged" as unhandled

It should also do the same thing as before on `MemberRemoved`, because a member might be removed without leaving. 

Another thing we should consider is if it would still be interesting to use the leaving/exiting nodes for the broadcast `Publish` messages, but not for the single destination `Send` messages.